service/controller/clusterapi/v17/resources/clusterid/create.go

```diff
@@ -2,8 +2,65 @@ package clusterid
 
 import (
 	"context"
+	"reflect"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	cmav1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring cluster status has cluster ID")
+
+	currentStatus := r.commonClusterStatusAccessor.GetCommonClusterStatus(cr)
+
```

Comment:
Isn't it more straightforward to check ID and return early?

```
if currentStatus.ID != "" {
	r.logger.LogCtx(ctx, "level", "debug", "message", "ensured cluster status has cluster ID")
	return nil
}

Reply:
Refactored.